### PR TITLE
Add easier PG setup for local development

### PR DESCRIPTION
**Pull request by ZJvandeWeg**
Recommend reviewing commit by commit, but: PG development is harder than it should be. This is an iteration trying to do better.

### CODE REVIEW

File: lib/commands/init.js [hunk 1]
```diff
@@ -67,6 +68,35 @@ async function run(options) {
             return
         }
     }
+
+	if (usePGDatabase(options.rootDir)) {
+		log(chalk.bold('Setting up PostgreSQL'))
+
+		try {
+			await runner("initdb", `initdb --pgdata=${path.join(options.rootDir, "data", "pg")}`)
+            log(`${chalk.greenBright('+')} initdb`)
```
Comment (Pezmc) [inline lib/commands/init.js]: The indenting as gone a bit funky here!

File: README.md [hunk 2]
```diff
@@ -148,4 +148,30 @@ find . -name .git -type d -execdir git status \;
...
+
+## Auxiliary services
+
+### PostgreSQL
+
+By default FlowForge uses SQLite for development. Given production systems do not
+tend to use SQLite but PostgreSQL (PG) it's advised to run PG for development too.
+As prerequisite, one should install PG on their own system.
+
+To use PG as development database ensure `packages/flowforge/etc/flowforge.yml`
```
Comment (Pezmc) [inline README.md:161]:
```suggestion
To use PG as development database ensure `packages/flowforge/etc/flowforge.local.ym`
```

File: README.md [hunk 3]
```diff
@@ -148,4 +148,30 @@ find . -name .git -type d -execdir git status \;
...
+has `postgres` set as database type. The host can be set to an **absolute** path
```
Comment (Pezmc) [inline README.md:162]:
```suggestion
has `postgres` set as database type. The host must be set to an **absolute** path
```

**Comment by Pezmc** [comment]
@ZJvandeWeg I've tried this out locally and hit a couple of snags (below), all fixed in the commits above with doc updates suggestions as comments on the PR.

- The script failed if data/pg didn't exist only the `data/.gitkeep` is set
- The script was reading from `flowforge.yml` but best practice to use `flowforge.local.yml instead`
- The file ended up with a mix of tabs and spaces, converted back to spaces

**Comment by ZJvandeWeg** [comment]
> The script failed if data/pg didn't exist only the data/.gitkeep is set

@Pezmc That's interesting: 

```
zegerjan@zeger-jans-air handbook % initdb --pgdata=/tmp/testdb
The files belonging to this database system will be owned by user "zegerjan".
This user must also own the server process.

<snip>

zegerjan@zeger-jans-air handbook % echo $?
0
```

**Comment by Pezmc** [comment]
I think it was the `readdirSync(pgDir)` rather than the initdb that failed!

File: README.md [hunk 4]
```diff
@@ -148,4 +148,30 @@ find . -name .git -type d -execdir git status \;
...
+  type: postgres
+  host: /path/to/flowforge-dev-env/data
+  port: 54321
+```
+
+Than run `npm run init` to setup your database.
+
+Start the server by running `postgres -s -D ./data/pg -k $(pwd)/data -p 54321`
+from the root directory, and keep the terminal window open. Just once you'll
+need to run `createdb -h $(pwd)/data -p 54321 flowforge`.
```
Comment (hardillb) [inline README.md:177]: This works, but it's not as clear as it could be, took me a moment to realise i needed 2 terminals

**Review by hardillb — APPROVED** [review]

**Pull request merged**